http/models.go

```diff
@@ -0,0 +1,19 @@
+package http
+
+// Params represents a generic map of key value pairs
+type Params map[string]string
+
+// Copy preserves the original params value (key value pair)
+// but stores the data in a different reference address
+func (originalParams Params) Copy() Params {
+	if originalParams == nil {
+		return nil
+	}
+
+	params := Params{}
+	for key, value := range originalParams {
+		params[key] = value
+	}
+
+	return params
+}
```

Comment:
Its not immediately obvious what this is related to - it doesn't seem specific to HTTP. I can see that you have brought it across, and I guess its used within the context of HTTP. Is there a better package / repo that we can move this to? (not a requested change as part of this PR, just a question to consider :) )

Reply:
I agree, I just copied the code across from go-ns/commons. I checked usages in other services, and found the following:
This  struct ( originally `common.Params` under go-ns), is used to create auditParams, sent to api.auditor.Record(...) by the following services:
- dp-search-api 
- dp-filter-api
- dp-dataset-api

I see that the audit logic lives under `go-ns/audit`, maybe we should migrate this file to a new  `dp-audit` service, once we decide to move go-ns/audit to its own repo?

Reply:
I will add this as a comment in the file - so that we remember in the future.

Reply:
So yes there will be a dp-audit library in the future; im currently working on a spike for this new library